controller/nc-news.controller.js

```diff
@@ -55,4 +56,27 @@ function getAllArticles(req, res, next) {
     .catch(next);
 }
 
-module.exports = { getApi, getTopics, getArticleById, getAllArticles };
+function getCommentsByArticeId(req, res, next) {
+  const { article_id } = req.params;
+  if (isNaN(Number(article_id))) {
+    handleError(400, "Invalid article id format", next);
+    return;
+  }
+  fetchCommentsByArticleId(article_id)
+    .then((comments) => {
+      if (!comments || comments.length === 0) {
```

Comment:
If the comments array is empty there are 2 possible reasons:
- article exists but has no comments.
- article does not exist.
It might be worth checking if the article exists to find our why the array is empty.